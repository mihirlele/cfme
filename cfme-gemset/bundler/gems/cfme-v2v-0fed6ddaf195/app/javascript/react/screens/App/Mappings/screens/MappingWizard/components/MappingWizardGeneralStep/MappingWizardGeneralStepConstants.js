export const SET_V2V_EDITING_MAPPING = 'SET_V2V_EDITING_MAPPING';
export const FETCH_V2V_CONVERSION_HOSTS = 'FETCH_V2V_CONVERSION_HOSTS';
