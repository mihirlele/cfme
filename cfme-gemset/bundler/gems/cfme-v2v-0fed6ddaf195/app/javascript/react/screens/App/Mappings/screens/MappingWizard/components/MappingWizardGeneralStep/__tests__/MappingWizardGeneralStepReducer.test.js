import generalStepReducer, { initialState } from '../MappingWizardGeneralStepReducer';
import { SET_V2V_EDITING_MAPPING, FETCH_V2V_CONVERSION_HOSTS } from '../MappingWizardGeneralStepConstants';
import { conversionHosts } from '../mappingWizardGeneralStep.fixtures';

it('sets default state', () => {
  const action = { type: '@@INIT' };
  const state = generalStepReducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it('sets editingMapping', () => {
  const action = {
    type: SET_V2V_EDITING_MAPPING,
    payload: '1'
  };
  const state = generalStepReducer(initialState, action);
  expect(state).toMatchSnapshot();
});

describe('fetching conversion hosts', () => {
  it('is pending', () => {
    const action = {
      type: `${FETCH_V2V_CONVERSION_HOSTS}_PENDING`
    };
    const prevState = initialState.set('isRejectedConversionHosts', true);
    const state = generalStepReducer(prevState, action);
    expect(state).toMatchSnapshot();
  });

  it('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_CONVERSION_HOSTS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingConversionHosts', true);
    const state = generalStepReducer(prevState, action);
    expect(state).toMatchSnapshot();
  });

  describe('is successful', () => {
    const prevState = initialState.set('isRejectedConversionHosts', true).set('isFetchingConversionHosts', true);

    test('and there is data', () => {
      const payload = {
        data: {
          ...conversionHosts
        }
      };
      const action = {
        type: `${FETCH_V2V_CONVERSION_HOSTS}_FULFILLED`,
        payload
      };
      const state = generalStepReducer(prevState, action);
      expect(state.conversionHosts).toHaveLength(3);
    });

    test('and there is no data', () => {
      const payload = {
        data: {
          ...conversionHosts,
          resources: []
        }
      };
      const action = {
        type: `${FETCH_V2V_CONVERSION_HOSTS}_FULFILLED`,
        payload
      };
      const state = generalStepReducer(prevState, action);
      expect(state).toMatchSnapshot();
    });
  });
});
