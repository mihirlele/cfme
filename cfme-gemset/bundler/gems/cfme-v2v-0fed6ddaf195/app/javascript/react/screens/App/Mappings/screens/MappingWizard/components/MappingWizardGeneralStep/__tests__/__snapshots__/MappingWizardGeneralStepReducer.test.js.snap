// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetching conversion hosts is pending 1`] = `
Object {
  "conversionHosts": Array [],
  "editingMapping": null,
  "errorFetchingConversionHosts": null,
  "isFetchingConversionHosts": true,
  "isRejectedConversionHosts": false,
}
`;

exports[`fetching conversion hosts is rejected 1`] = `
Object {
  "conversionHosts": Array [],
  "editingMapping": null,
  "errorFetchingConversionHosts": "error",
  "isFetchingConversionHosts": false,
  "isRejectedConversionHosts": true,
}
`;

exports[`fetching conversion hosts is successful and there is no data 1`] = `
Object {
  "conversionHosts": Array [],
  "editingMapping": null,
  "errorFetchingConversionHosts": null,
  "isFetchingConversionHosts": false,
  "isRejectedConversionHosts": false,
}
`;

exports[`sets default state 1`] = `
Object {
  "conversionHosts": Array [],
  "editingMapping": null,
  "errorFetchingConversionHosts": null,
  "isFetchingConversionHosts": false,
  "isRejectedConversionHosts": false,
}
`;

exports[`sets editingMapping 1`] = `
Object {
  "conversionHosts": Array [],
  "editingMapping": "1",
  "errorFetchingConversionHosts": null,
  "isFetchingConversionHosts": false,
  "isRejectedConversionHosts": false,
}
`;
